NodeInstaller/index.js

```diff
@@ -1,248 +1,255 @@
-                            }
-                        }
-                    });
-
-                } else {
-                    console.log("Something went wrong. Please try again.");
-                    process.exit();
-                }
-            });
-
-        } else {
-            console.log("Discord resources not found at: " + _discordPath);
-            process.exit();
-        }
-
-    });
-
-}
-
-function init() {
-
-    console.log("BetterDiscord Simple Installer v0.3 for Discord "+dver+" by Jiiks.");
-    console.log("If Discord has updated then download the latest installer.");
-
-    var rl = readline.createInterface({ input: process.stdin, output: process.stdout });
-
-    if (_force == false) {
-        rl.question("The following directories will be deleted if they exists: discorpath/app, discordpath/node_modules/BetterDiscord, is this ok? Y/N", function(answer) {
-
-            var alc = answer.toLowerCase();
-
-            switch(alc) {
-                case "y":
-                    install();
-                    break;
-                case "yes":
-                    install();
-                    break;
-                case "n":
-                    process.exit();
-                    break;
-                case "no":
-                    process.exit();
-                    break;
-            }
-        });
-    } else {
-        install();
-    }
-}
-
-init();
+/*
+ * BetterDiscordApp Installer v0.3.2
+ */
+
+var dver = "0.0.284";
+
+var asar = require('asar');
+var wrench = require('wrench');
+var fs = require('fs');
+var readline = require('readline');
+var util = require('util');
+
+var _importSplice;
+var _functionSplice;
+var _functionCallSplice;
+
+var _discordPath;
+var _appFolder = "/app";
+var _appArchive = "/app.asar";
+var _packageJson = _appFolder + "/package.json";
+var _index = _appFolder + "/app/index.js";
+
+var _force = false;
+
+// Get Arguments
+process.argv.forEach(function (val, index, array) {
+    if (val == "--force") {
+        _force = true;
+    }
+
+    if (val == "-d" || val == "--directory") {
+        _discordPath = array[(index+1)]
+    }
+});
+
+function install() {
+
+    if (typeof _discordPath == 'undefined') {
+        var _os = process.platform;
+        if (_os == "win32") {
+            _importSplice = 89;
+            _functionCallSplice = 497;
+            _functionSplice = 601;
+           _discordPath = process.env.LOCALAPPDATA + "/Discord/app-"+dver+"/resources";
+        } else if (_os == "darwin") {
+            _importSplice = 67;
+            _functionCallSplice = 446;
+            _functionSplice = 547;
+            _discordPath = "/Applications/Discord.app/Contents/Resources"; // Defaults to Applications directory
+        } else if (_os == "linux") {
+			_importSplice = 63;
+			_functionCallSplice = 507;
+			_functionSplice = 580;
+			_discordPath = "/opt/DiscordCanary/resources";
+			// not sure where the .deb install, and most people are going to 
```

Comment:
The original required the user to install in /opt, and not everyone is on Debian or Ubuntu, so I'm inclined to leave that for now.